File: crates/klyron_docker/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_docker {

DockerBuilder::DockerBuilder() : config_{} {}

DockerBuilder& DockerBuilder::set_config(const DockerConfig& cfg) {
    config_ = cfg;
    return *this;
}

DockerConfig DockerBuilder::config() const {
    return config_;
}

}

File: crates/klyron_docker/bindings/rust/builder.rs
//! Builder for klyron_docker
use crate::types::DockerConfig;

pub struct DockerBuilder {
    config: DockerConfig,
}

impl DockerBuilder {
    pub fn new() -> Self {
        Self { config: DockerConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> DockerConfig {
        self.config
    }
}
